### PR TITLE
fix: deprecate axe.commons.utils namespace

### PR DESCRIPTION
**Pull request by jkodu**
This PR does the below:
- Migrated utility functions from `axe.commons.utils` namespace to `axe.utils`

Why?
- There were a plethora of namespace pollution - eg: `var utils = (commons.utils = axe.utils);`

Closes issue: NA

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Has documentation updated, a DU ticket, or requires no documentation change
- [x] Includes new tests, or was unnecessary
- [x] Code is reviewed for security by: Stephen

### CODE REVIEW

File: lib/commons/utils/index.js [hunk 1]
```diff
@@ -1,10 +0,0 @@
-/* exported utils */
-/* global axe */
-/*eslint no-unused-vars: 0*/
-/**
- * Namespace for general utilities.
- * @namespace utils
- * @memberof axe.commons
- */
-
-var utils = (commons.utils = axe.utils);
```
Comment (WilcoFiers) [inline lib/commons/utils/index.js]: We need to keep this around. Removing this would be a breaking change.

File: test/commons/utils/index.js [hunk 2]
```diff
@@ -1,21 +0,0 @@
...
-		assert.isFunction(axe.commons.utils.matchesSelector);
-	});
-});
-describe('utils.clone', function() {
-	'use strict';
-
-	it('should be a function', function() {
-		assert.isFunction(axe.commons.utils.clone);
-	});
-});
```
Comment (WilcoFiers) [inline test/commons/utils/index.js]: Same here, we don't want to make a breaking change. Just check that `assert.equal(axe.utils, axe.commons.utils)`.

File: lib/commons/utils/index.js (hunk 1 above)
Comment (stephenmathieson) [reply lib/commons/utils/index.js]: What would this break?

**Comment by stephenmathieson** [comment]
@WilcoFiers I don't understand why removing the aliased namespace would be a breaking change. From what I can tell, it's not part of any public API. Can you please clarify?

File: lib/commons/utils/index.js (hunk 1 above)
Comment (WilcoFiers) [reply lib/commons/utils/index.js]: Commons are pseudo-public. They are used in custom rules which clients use. We have to be careful about changing commons.

**Pull request merged**